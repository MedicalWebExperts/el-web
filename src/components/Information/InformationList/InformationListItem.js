import React from 'react';
import { string, shape } from 'prop-types';
import css from 'styled-jsx/css';

import theme from '../../../styles';

const propTypes = {
  text: string.isRequired,
  styles: shape({}),
};

const defaultProps = {
  styles: {},
};

const style = css`
  li {
    padding: 0 0 0 20px;
    position: relative;
    color: ${theme.colors.default};
    font-family: ${theme.font.primary};
    font-size: ${theme.font.size}px;
    line-height: 1.6;
  }
  li:before {
    content: '';
    position: absolute;
    left: 0;
    top: 7px;
    right: auto;
    width: 8px;
    height: 8px;
    border-bottom: solid 1px currentColor;
    border-left: solid 1px currentColor;
    transform: rotate(-135deg);
    color: #4a90e2;
  }
`;

const InformationListItem = ({ text, styles }) => (
  <li style={styles}>
    <style jsx>{style}</style>
    {text}
  </li>
);

InformationListItem.propTypes = propTypes;
InformationListItem.defaultProps = defaultProps;
export default InformationListItem;
