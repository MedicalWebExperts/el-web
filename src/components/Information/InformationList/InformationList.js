import React from 'react';
import { arrayOf, shape, string } from 'prop-types';
import css from 'styled-jsx/css';

import InformationListItem from './InformationListItem';

const propTypes = {
  data: arrayOf(string).isRequired,
  styles: shape({}),
};

const defaultProps = {
  styles: {},
};

const style = css`
  ul {
    list-style: none;
    padding: 0;
    margin: 0;
  }
`;

const InformationList = ({ data, styles }) => (
  <ul style={styles}>
    <style jsx>{style}</style>

    {data.map((item, index) => (
      <InformationListItem text={item} key={index} />
    ))}
  </ul>
);

InformationList.propTypes = propTypes;
InformationList.defaultProps = defaultProps;
export default InformationList;
