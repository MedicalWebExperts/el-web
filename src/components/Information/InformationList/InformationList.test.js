import React from 'react';
import renderer from 'react-test-renderer';
import InformationList from './InformationList';

const defaultProps = {
  data: [
    'American Medical Association',
    'Texas Medical Association',
    'American Academy of Family Physicians',
  ],
};

const customStyles = {
  styles: { margin: 20 },
};

describe('Component InformationList Snapshot', () => {
  it('without arguments, take the default values', () => {
    const tree = renderer.create(<InformationList {...defaultProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('using custom styles', () => {
    const tree = renderer
      .create(<InformationList {...Object.assign({}, defaultProps, customStyles)} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
