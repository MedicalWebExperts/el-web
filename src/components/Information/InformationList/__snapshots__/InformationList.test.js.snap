// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component InformationList Snapshot using custom styles 1`] = `
<ul
  className="jsx-3401328911"
  style={
    Object {
      "margin": 20,
    }
  }
>
  <li
    className="jsx-1051720404"
    style={Object {}}
  >
    American Medical Association
  </li>
  <li
    className="jsx-1051720404"
    style={Object {}}
  >
    Texas Medical Association
  </li>
  <li
    className="jsx-1051720404"
    style={Object {}}
  >
    American Academy of Family Physicians
  </li>
</ul>
`;

exports[`Component InformationList Snapshot without arguments, take the default values 1`] = `
<ul
  className="jsx-3401328911"
  style={Object {}}
>
  <li
    className="jsx-1051720404"
    style={Object {}}
  >
    American Medical Association
  </li>
  <li
    className="jsx-1051720404"
    style={Object {}}
  >
    Texas Medical Association
  </li>
  <li
    className="jsx-1051720404"
    style={Object {}}
  >
    American Academy of Family Physicians
  </li>
</ul>
`;
